### ISSUE TITLE
precedence has false positives through macros

### ISSUE THREAD

**Issue by gnzlbg** [rust-lang-nursery/rust-clippy#2323]
[Minimal working example](https://play.rust-lang.org/?gist=f0d4ed7b0d7b28a82d642c80abca7ce4&version=stable): 

```rust
macro_rules! trip {
   ($a:expr) => { 
    match $a & 0b1111_1111u8 {
        0 => println!("a is zero ({})", $a),
        _ => println!("a is {}", $a),
    }
   };
}

fn main() {
    let b = 3;
    trip!(b * 8);
}
```

produces

```shell
  Compiling playground v0.0.1 (file:///playground)
warning: operator precedence can trip the unwary
  --> src/main.rs:3:11
   |
3  |     match $a & 0b1111_1111u8 {
   |           ^^^^^^^^^^^^^^^^^^ help: consider parenthesizing your expression: `(b * 8) & 0b1111_1111u8`
...
12 |     trip!(b * 8);
   |     ------------- in this macro invocation
   |
   = note: #[warn(precedence)] on by default
   = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.177/index.html#precedence
```

**Comment by oli-obk** [comment]
just needs a macro check.
